luaui/configs/gridmenu_config.lua

```diff
@@ -23,175 +21,159 @@ local categories = {
-			for row =1,3 do
-				for col =1,4 do
+		local uncategorizedCount = 0
+		for cat = 1, 4 do
+			for row = 1, 3 do
+				for col = 1, 4 do
 					local unitAtPos = ugrid[cat] and ugrid[cat][row] and ugrid[cat][row][col]
-
 					if unitAtPos then
 						local unit = UnitDefNames[unitAtPos]
-
 						if unit and unit.id and builderCanBuild[unit.id] then
-							gridPosUnit[builderId][cat .. row .. col] = unit.id
-							unitGridPos[builderId][cat][unit.id] = cat .. row .. col
-							hasUnitGrid[builderId][unit.id] = true
-							uncategorizedCount = uncategorizedCount + 1
-							homeGridPos[builderId][cat][uncategorizedCount] = unit.id
+							gridPosUnit [builderId][cat .. row .. col] = unit.id
+							unitGridPos [builderId][cat][unit.id] = cat .. row .. col
+							hasUnitGrid [builderId][unit.id] = true
+							uncategorizedCount            = uncategorizedCount + 1
+							homeGridPos  [builderId][cat][uncategorizedCount] = unit.id
 						end
 					end
 				end
 			end
-			uncategorizedCount = 0;
+			uncategorizedCount = 0
 		end
 	end
 end
 
 for uname, ugrid in pairs(labGrids) do
 	local udef = UnitDefNames[uname]
 	if not udef then
-		Spring.Echo('gridmenu config: no unitdefname found for: '..uname)
+		Spring.Echo('gridmenu config: no unitdefname found for: ' .. uname)
 	else
 		local uid = udef.id
 
 		unitGridPos[uid] = {}
 		gridPosUnit[uid] = {}
-		local uCanBuild = {}
 
-		local uBuilds = udef.buildOptions
-		for i = 1, #uBuilds do
-			uCanBuild[uBuilds[i]] = true
+		local uCanBuild = {}
+		for _, u in ipairs(udef.buildOptions) do
+			uCanBuild[u] = true
 		end
 
-		for r=1,3 do
-			for c=1,4 do
+		for r = 1, 3 do
+			for c = 1, 4 do
 				local index = (r - 1) * 4 + c
-				local ugdefname = ugrid[index]
-
-				if ugdefname then
-					local ugdef = UnitDefNames[ugdefname]
-
-					if ugdef and ugdef.id and uCanBuild[ugdef.id] then
-						gridPosUnit[uid][r .. c] = ugdef.id
-						unitGridPos[uid][ugdef.id] = r .. c
+				local ugdefName  = ugrid[index]
+				if ugdefName then
+					local ugDef = UnitDefNames[ugdefName]
+					if ugDef and ugDef.id and uCanBuild[ugDef.id] then
+						gridPosUnit [uid][r .. c]  = ugDef.id
+						unitGridPos [uid][ugDef.id] = r .. c
 					end
 				end
 			end
 		end
 	end
 end
 
-
 for unitDefID, unitDef in pairs(UnitDefs) do
 	unitCategories[unitDefID] = categoryGroupMapping[unitDef.customParams.unitgroup] or BUILDCAT_UTILITY
 end
 
-
+---------------------------------------------------------------- small utilities
 local function getCategoryIndex(category)
-	if category == BUILDCAT_ECONOMY then return 1
-	elseif category == BUILDCAT_COMBAT then return 2
-	elseif category == BUILDCAT_UTILITY then return 3
-	elseif category == BUILDCAT_PRODUCTION then return 4
-	else return nil end
+	if     category == BUILDCAT_ECONOMY    then return 1
+	elseif category == BUILDCAT_COMBAT     then return 2
+	elseif category == BUILDCAT_UTILITY    then return 3
+	elseif category == BUILDCAT_PRODUCTION then return 4 end
```

Comment:
Why is this a function? Static data should be stored in a table. The constants should also be assigned to these integer values, not the mutable translated string names.